MatchInfoModule: Fixes and working version

KamilKhairullin opened the pull request.

delet-dis committed:
Merge pull request #105 from DotaSirius/issue-104-IE

🔧 fixed tests

KamilKhairullin committed:
MatchInfoModule: fixes

KamilKhairullin committed:
Merge branch 'main' into issue-30-kh-2

# Conflicts:
#	DotaStats.xcodeproj/project.pbxproj

KamilKhairullin committed:
Merge branch 'main' into issue-30-kh-2

KamilKhairullin committed:
MatchInfoModule: fixes on pull request

KamilKhairullin committed:
Merge branch 'main' into issue-30-kh-2

KamilKhairullin committed:
MatchInfoModule: Fixes and working version

KamilKhairullin committed:
MatchInfoModule: Fixes and working version

KamilKhairullin committed:
Merge branch 'issue-30-kh-2' of https://github.com/DotaSirius/Dota2_Sirius into issue-30-kh-2

KamilKhairullin committed:
MatchInfoModule: Fixes and working version

Antowkos commented on DotaStats/AppCoordinator.swift:
```diff
@@ -58,6 +58,17 @@ extension AppCoordinator {
         )
     }
+
+    private func matchInfoModuleBuilder() -> MatchInfoModuleBuilder {
```
Фабричные методы лучше всего начинать со слова `make`
```suggestion
    private func makeMatchInfoModuleBuilder() -> MatchInfoModuleBuilder {
```

Antowkos commented on DotaStats/AppCoordinator.swift:
```diff
@@ -79,6 +92,8 @@ extension AppCoordinator: SearchPlayerModuleOutput {
 
+extension AppCoordinator: MatchInfoModuleOutput {}
+
 final class StubImageNetworkService: ImageNetworkService {
```
А что здесь делает стабовый сервис? Он все еще нужен?

KamilKhairullin replied:
Используется в другом модуле. Для нашего не нужен, но без него не запускается. Так что вопрос, скорее всего, к тем, кто писал модуль SearchPlayerModule

Antowkos commented on DotaStats/Modules/MatchInfoModule/MatchInfoModuleBuilder.swift:
```diff
@@ -0,0 +1,20 @@
+    }
+
+    init(output: MatchInfoModuleOutput, networkService: MatchDetailService, converter: MatchInfoConverter) {
+        self.converter = converter
```
Зачем сохраняешь конвертер в билдере?

Antowkos commented on DotaStats/Modules/MatchInfoModule/Presenter/MatchInfoConverter.swift:
```diff
@@ -0,0 +1,241 @@
+    }
+}
+
+private extension MatchInfoConverterImp {
```
Вот лучше бы сделать наоборот. Сам конвертер умеет что-то конвертировать с помощью приватных функций и есть `extension`, в котором поддерживается протокол `MatchInfoConverter`

Antowkos commented on DotaStats/Modules/MatchInfoModule/Presenter/MatchInfoConverter.swift:
```diff
@@ -0,0 +1,241 @@
+import UIKit
+
+protocol MatchInfoConverter: AnyObject {
+    func mainMatchInfo(rawMatchInfo: MatchDetail) -> MainMatchInfo
```
С учетом того, что делает метод лучше бы его сигнатурку подправить до такой
```suggestion
    func mainMatchInfo(from rawMatchInfo: MatchDetail) -> MainMatchInfo
```
Для всех остальных подобных методов в протоколе можно сделать тоже самое

Antowkos commented on DotaStats/Modules/MatchInfoModule/Presenter/MatchInfoModulePresenter.swift:
```diff
@@ -0,0 +1,122 @@
+
+    private func updateView() {
+        state = .loading
+        _ = networkService.requestMatchDetail(id: matchId) { [weak self] result in
```
Тут либо надо сохранять токен запроса и отменять его при повторном вызове, либо убрать `_ = `

KamilKhairullin replied:
> Тут либо надо сохранять токен запроса и отменять его при повторном вызове, либо убрать `_ =`

Без `_ =` всплывает warning 
Result of call to 'requestMatchDetail(id:completion:)' is unused

Antowkos replied:
Почитай про `@discardableResult`

Antowkos commented on DotaStats/Modules/MatchInfoModule/Presenter/MatchInfoModulePresenter.swift:
```diff
@@ -0,0 +1,122 @@
+
+            case .failure(let error):
+                self.state = .error
+                print(error)
```
Принтить ошибки можно во время дебага, но пушить их в main без грамотной обработки в продакшене не стоит.

Antowkos commented on DotaStats/Modules/MatchInfoModule/Presenter/MatchInfoModulePresenter.swift:
```diff
@@ -0,0 +1,122 @@
+extension MatchInfoModulePresenter: MatchInfoModuleInput {
+    func setMatchId(_ id: Int) {
+        matchId = id
+        updateView()
```
Кажется, что этот код вызывается до того, как `view` отображается на экране. Не будет ли здесь каких-нибудь проблем?

KamilKhairullin replied:
> Кажется, что этот код вызывается до того, как `view` отображается на экране. Не будет ли здесь каких-нибудь проблем?

Проблем возникнуть не должно. Тут скорее неправильное название метода. Метод делает запрос данных из сети и после этого обновляет переменную state. А во view данные уходят в didSet переменной state, предварительно законверченные. Я думаю, стоит переименовать метод.

Antowkos commented on DotaStats/Modules/MatchInfoModule/View/AdditionalMatchInfoTableViewCell.swift:
```diff
@@ -0,0 +1,140 @@
+    override init(style: UITableViewCell.CellStyle, reuseIdentifier: String?) {
+        super.init(style: style, reuseIdentifier: reuseIdentifier)
+        setup()
+        setUpConstraints()
```
Пжлст, давай `setup` писать однообразно. Посмотри по всему коду
```suggestion
        setupConstraints()
```

NickaKos11 replied:
сделала

Antowkos commented on DotaStats/Modules/MatchInfoModule/View/AdditionalMatchInfoTableViewCell.swift:
```diff
@@ -0,0 +1,140 @@
+    }
+
+    func setUpConstraints() {
+        [matchIdCopyButton,
+         regionStackView,
+         skillStackView].forEach {
+            $0.translatesAutoresizingMaskIntoConstraints = false
+        }
```
Перенеси в настройки самих вьюх, пж

NickaKos11 replied:
сделала

Antowkos commented on DotaStats/Modules/MatchInfoModule/View/AdditionalMatchInfoTableViewCell.swift:
```diff
@@ -0,0 +1,140 @@
+        }
+
+        NSLayoutConstraint.activate([
+            matchIdCopyButton.topAnchor.constraint(equalTo: contentView.topAnchor, constant: 16),
```
лучше циферку 16 вынести в отдельную переменную, обозвать как-то получше, например `inset` и везде юзать. Посмотри везде, пж

NickaKos11 replied:
сделала

Antowkos commented on DotaStats/Modules/MatchInfoModule/View/MatchInfoViewController.swift:
```diff
@@ -0,0 +1,106 @@
+            spiner.startAnimating()
+        case .error:
+            spiner.removeFromSuperview()
+        // TO DO: Errors
```
```suggestion
        // TODO: Errors
```

Antowkos commented on DotaStats/Modules/MatchInfoModule/View/MatchInfoViewController.swift:
```diff
@@ -0,0 +1,106 @@
+            spiner.removeFromSuperview()
+        // TO DO: Errors
+        case .success:
+            view.addSubview(tableView)
+            setUpConstraints()
```
На каждый `.success` будет добавляться таблица

KamilKhairullin committed:
Пофикшена часть комментариев.

Остались нефикшенными:
1. MatchInfoModulePresenter.swift - line 80 - тут либо надо сохранять токен запроса и отменять его при повторном вызове, либо убрать _ = (Вылезает warning, если убрать "_ =" )
2. AdditionalMatchInfoTableViewCell.swift - line 105 - Перенеси в настройки самих вьюх, пж (Сделает Вероника)
3. AdditionalMatchInfoTableViewCell.swift - all file - лучше циферку 16 вынести в отдельную переменную, обозвать как-то получше, например inset и везде юзать. Посмотри везде, пж (Вероника)
4. MatchInfoViewController.swift - line 102 - На каждый .success будет добавляться таблица (Как пофиксить?)

NickaKos11 committed:
fixed comments

NickaKos11 committed:
fixed warnings

NickaKos11 committed:
add error handling

Antowkos commented on DotaStats/Modules/MatchInfoModule/Model/MatchesInfoModuleState.swift:
```diff
@@ -1,5 +1,5 @@
     case success
     case loading
-    case error
+    case error(HTTPError)
```
Лучше бы эту штуку конвертировать в удобоваримую ошибку для вьюхи

Antowkos commented on DotaStats/Modules/MatchInfoModule/View/AdditionalMatchInfoTableViewCell.swift:
```diff
@@ -92,6 +98,7 @@ final class AdditionalMatchInfoTableViewCell: UITableViewCell {
     @objc
     private func copyMatchIdButtonPressed() {
         UIPasteboard.general.string = matchID
+        matchIdCopyButton.setTitle("Copied!", for: .normal)
```
Будет здорово, если спустя какое-то время (например, 5 секунд) заголовок кнопки вернется в свое первоначальное состояние

Antowkos commented on DotaStats/Modules/MatchInfoModule/View/MatchInfoViewController.swift:
```diff
@@ -25,10 +27,21 @@ final class MatchInfoViewController: UIViewController {
+        view.alpha = 0
+        view.translatesAutoresizingMaskIntoConstraints = false
+        view.isUserInteractionEnabled = true
+        let tapActionHideError = UITapGestureRecognizer(target: self, action: #selector(handleTap(_:)))
```
Сейчас название этой переменной говорит о том, что в ней хранится ошибка. Лучше назвать ее примерно так
```suggestion
        let tapActionGestureRecognizer = UITapGestureRecognizer(target: self, action: #selector(handleTap(_:)))
```

Antowkos commented on DotaStats/Modules/MatchInfoModule/View/MatchInfoViewController.swift:
```diff
@@ -42,18 +55,58 @@ final class MatchInfoViewController: UIViewController {
+        }
+    }
+
+    @objc func handleTap(_: UITapGestureRecognizer) {
```
Не самое лучшее название для функции. По имени этой функции кажется, что внутри нее обрабатываются любые касания, а это не так. Надо в названии отразить что конкретно она делает

KamilKhairullin committed:
fixes of presenter & error handling

KamilKhairullin committed:
fixes of presenter & error handling

Antowkos reviewed (commented):
Про возврат надписи при копировании не забудьте, пжлст

KamilKhairullin committed:
Merge branch 'issue-30-kh-2' of https://github.com/DotaSirius/Dota2_Sirius into issue-30-kh-2
